src/core/mac/sub_mac.cpp

```diff
@@ -356,7 +356,7 @@ void SubMac::StartCsmaBackoff(void)
 
         if (ShouldHandleTransmitTargetTime())
         {
-            if (otPlatRadioGetNow(&GetInstance()) <
+            if (static_cast<uint32_t>(otPlatRadioGetNow(&GetInstance())) <
```

Comment:
```suggestion
            if (static_cast<int32_t>(static_cast<uint32_t>(otPlatRadioGetNow(&GetInstance())) - static_cast<uint32_t>(mTransmitFrame.mInfo.mTxInfo.mTxDelayBaseTime + mTransmitFrame.mInfo.mTxInfo.mTxDelay)) < 0)
```

Reply:
What about:
```
if (static_cast<uint32_t>(otPlatRadioGetNow(&GetInstance()) < 
    static_cast<uint32_t>(mTransmitFrame.mInfo.mTxInfo.mTxDelayBaseTime + mTransmitFrame.mInfo.mTxInfo.mTxDelay))
```
Is there a difference between this and your suggection?

Reply:
Will `mTransmitFrame.mInfo.mTxInfo.mTxDelayBaseTime + mTransmitFrame.mInfo.mTxInfo.mTxDelay` overflow and wrap?

Reply:
Use `Time` operator to compare now.